### DISCUSSION TITLE
Is there a way to install in MacOS without root password?

### DISCUSSION THREAD

**Discussion by alcarraz** [DevToys-app/DevToys#1313] [Q&A] [answered]
Hi there, I want to try this out, but when I tried to install in my work machine using brew. It asked me for an admin password. 

Is there a recommended way to do that in user space?

Thank you.

**Comment by veler** [comment]
Hi,
That is quite interesting. It never asked me a password to install it. I do not know why it's asking you one.

**Reply by alcarraz** [reply]
Thank you for your prompt response, here is the output:

```
==> Downloading https://github.com/DevToys-app/DevToys/releases/download/v2.0.4.0/devtoys_osx_x64.zip
==> Downloading from https://objects.githubusercontent.com/github-production-release-asset-2e65be/411493648/60dfbb0c-461b-4f58-886e-9ce3698dd2f3?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releasea
#################################################################################################################################################################################################### 100.0%
==> Installing Cask devtoys
==> Moving App 'DevToys.app' to '/Applications/DevToys.app'
Password:
```

I don't have permission on `/Applications`, is there a way to make brew install to other paths?

I have this problem with many other recipes, but not all.

**Reply by veler** [reply, marked as answer]
Ah, I see. So it's specific to your machine configuration. You can probably go to https://devtoys.app and download manually the application bundle, without using Homebrew, then move it wherever you want. You will not have automatic update though.

**Reply by alcarraz** [reply]
Well, to mine and any restricted machine where the user can almost only modify their home directory. But yes, this is a common issue with graphical apps I want to install with homebrew.

I'm new to Mac, and I'm hating it. Most things, like homebrew, assume that it is a single user machine, and on top of that, that the single user can modify whatever it pleases them. And most devs rely on brew, or the store (from which I can't install also) to distribute their stuff. 

Yes, while I write this, I notice that the same may happen on Linux, and I may have not noticed it because I just have full privileges on my machine. So, I apologize for my (now) knowingly misplaced rant. But it is frustrating that I cannot auto update or even install most apps in the computer I spend most of my time without having to ask for permission.

**Reply by veler** [reply]
That's actually a fair rant. I'm new to Mac OS myself. I got a Mac Mini only for the purpose of porting DevToys on it. I had to go through a steep learning curve and obviously missed some things. I will try to see if it's possible to make homebrew installing the app in user's Applications folder instead of system on.

**Reply by veler** [reply]
Made an issue here: https://github.com/DevToys-app/DevToys/issues/1314

**Reply by alcarraz** [reply]
Thank you for your kindness, maybe it's a fair rant, but no against this particular project, and as a developer, I am part of the problem.

Being this a tool for developers, I may suggest an easy< although not elegant, mitigation. What if, as part of the repo, you provide an update script that checks the latest release, and then git checkouts the corresponding tag if there is a newer version than the current checked out, and then build the binary. To prevent from doing this on another branch, it would do that only if the current head corresponds to a release. This is just a wild suggestion that I hope it can trigger a smarter idea.

Thank you.

**Reply by veler** [reply]
Thanks for the suggestion. I don't think it will work though, for many reasons, but here are 2 main:
1. Once checked out all the code, I'd also need to bring all the tooling for building (.NET & NodeJS), which starts to add up, and risk to leave some clutter behind.
2. The app needs to be notarized / signed to run on Mac. It needs some secret information that for obvious reason should not be on the customer machine.

**Comment by alcarraz** [comment]
Ok, I was thinking like it was a Linux env, regarding point 2, Ididn't know
MacOS won't let me run stuff I compiled myself. But you are right regarding
point 1 anyway.

Thanks

----
Enviado desde mi móvil, disculpas por la brevedad.

Sent from my cellphone, I apologize for the brevity.

Andrés Alcarraz.
